### PR TITLE
fix: refactor blockaid metrics sync functions to async

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**
After this [PR](https://github.com/MetaMask/metamask-mobile/pull/10267) to use the Security alert API to get the supported chains via `isChainSupported` which now it's an async function and we saw that it is used across the Blockaid metrics in the mobile client. So this PR aims to refactor every place that uses it to support it asynchronously.

## **Related issues**

Fixes: https://github.com/MetaMask/mobile-planning/issues/1893
Related: https://github.com/MetaMask/mobile-planning/issues/1870

## **Manual testing steps**

1. Go to this test dapp and perform a regression test using "PPOM - Malicious Transactions and Signatures"
2. Check if metrics are going to Segment (the env `SEGMENT_WRITE_KEY` will be required to perform this test)

## **Screenshots/Recordings**

[test-ppom.webm](https://github.com/user-attachments/assets/423fae81-181b-4753-934e-55358fa088fc)

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by pedronfigueiredo — DISMISSED** [review]

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by pedronfigueiredo — DISMISSED** [review]

**Review by jpuri — CHANGES_REQUESTED** [review]
If a confirmation has `securityAlertResponse` we can add metrics for it. We do validation only after checking if blockaid is enabled. Check before getting metrics params again looks redundant.

**Comment by vinistevam** [comment]
> If a confirmation has `securityAlertResponse` we can add metrics for it. We do validation only after checking if blockaid is enabled. Check before getting metrics params again looks redundant.

Reverted changes in `getBlockaidMetricsParams` and `getBlockaidTransactionMetricsParams`.

**Review by matthewwalsh0 — DISMISSED** [review]

File: app/components/Views/confirmations/SendFlow/Confirm/index.js [hunk 1]
```diff
@@ -1285,7 +1289,7 @@ class Confirm extends PureComponent {
           layout="vertical"
         />
         <ScrollView style={baseStyles.flexGrow} ref={this.setScrollViewRef}>
-          {isBlockaidFeatureEnabled() && this.state.transactionMeta?.id && (
```
Comment (jpuri) [inline app/components/Views/confirmations/SendFlow/Confirm/index.js:1288]: We can even avoid `isBlockaidFeatureEnabled` below as `TransactionBlockaidBanner` render only is there is a `securityAlertResponse`.

File: app/components/Views/confirmations/SendFlow/Confirm/index.js [hunk 2]
```diff
@@ -474,7 +474,10 @@ class Confirm extends PureComponent {
 
     this.setState({ result, transactionMeta });
 
-    if (isBlockaidFeatureEnabled()) {
+    const isBlockaidEnabled = await isBlockaidFeatureEnabled();
+    this.setState({ isBlockaidFeatureEnabled: isBlockaidEnabled });
+
+    if (isBlockaidEnabled) {
```
Comment (jpuri) [inline app/components/Views/confirmations/SendFlow/Confirm/index.js:480]: We can avoid this check here as `ppomUtil.validateRequest(reqObject, id);` will work only if blockaid is enabled.

File: app/lib/ppom/ppom-util.ts [hunk 3]
```diff
@@ -64,14 +64,13 @@ async function validateRequest(req: PPOMRequest, transactionId?: string) {
...
-
   const chainId = NetworkController.state.providerConfig.chainId;
   const isConfirmationMethod = CONFIRMATION_METHODS.includes(req.method);
   const isSupportedChain = await isChainSupported(chainId);
-
+  const isBlockaidFeatEnabled = await isBlockaidFeatureEnabled();
   if (
     !ppomController ||
-    !isBlockaidFeatureEnabled() ||
+    !isBlockaidFeatEnabled ||
```
Comment (jpuri) [inline app/lib/ppom/ppom-util.ts:73]: 👍

**Review by jpuri — COMMENTED** [review]
Changes look great, I left some small feedbacks to be addressed.

File: app/components/Views/confirmations/SendFlow/Confirm/index.js (hunk 2 above)
Comment (vinistevam) [reply app/components/Views/confirmations/SendFlow/Confirm/index.js]: done [here](https://github.com/MetaMask/metamask-mobile/pull/10518/commits/8ee73b8a2d0ffac98af2a7bbcf5401cb88f10d39)

**Review by jpuri — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**